src/Compilers/Core/Portable/Collections/SmallDictionary.cs

```diff
@@ -62,14 +67,14 @@ private int GetHashCode(K k)
             return Comparer.GetHashCode(k);
         }
 
-        public bool TryGetValue(K key, out V value)
+        public bool TryGetValue(K key, [NotNullWhen(returnValue: true)] out V value)
```

Comment:
❕ At minimum, this needs `[MaybeNullWhen(false)]`. If the dictionary does not allow null values (as is suggested by `[NotNullWhen(true)]`), consider adding a generic constraint rather than using this attribute.